posthog/migrations/0206_global_tags_setup.py

```diff
@@ -0,0 +1,80 @@
+# Generated by Django 3.2.5 on 2022-02-10 01:05
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+import posthog.models.utils
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("posthog", "0205_auto_20220204_1748"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="Tag",
+            fields=[
+                (
+                    "id",
+                    models.UUIDField(
+                        default=posthog.models.utils.UUIDT, editable=False, primary_key=True, serialize=False
+                    ),
+                ),
+                ("name", models.CharField(max_length=255)),
+                ("team", models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to="posthog.team")),
+            ],
+        ),
+        migrations.CreateModel(
+            name="TaggedItem",
+            fields=[
+                (
+                    "id",
+                    models.UUIDField(
+                        default=posthog.models.utils.UUIDT, editable=False, primary_key=True, serialize=False
+                    ),
+                ),
+                (
+                    "action",
+                    models.ForeignKey(
+                        blank=True,
+                        null=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        related_name="tags",
+                        to="posthog.action",
+                    ),
+                ),
+                (
+                    "feature_flag",
+                    models.ForeignKey(
+                        blank=True,
+                        null=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        related_name="tags",
+                        to="posthog.featureflag",
+                    ),
+                ),
+                (
+                    "tag",
+                    models.ForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, related_name="taggeditems", to="posthog.tag"
+                    ),
+                ),
+            ],
+        ),
+        migrations.AddConstraint(
+            model_name="taggeditem",
+            constraint=models.CheckConstraint(
+                check=models.Q(
+                    models.Q(("action__isnull", True), ("feature_flag__isnull", True)),
+                    models.Q(("action__isnull", False), ("feature_flag__isnull", True)),
+                    models.Q(("action__isnull", True), ("feature_flag__isnull", False)),
+                    _connector="OR",
+                ),
+                name="at_most_one_related_object",
+            ),
+        ),
```

Comment:
I'm not convinced this makes sense at a database level. It feels like it'll just get needlessly complicated for little gain somehow 🤔